project_event/models/project_task.py

```diff
@@ -657,20 +657,28 @@ def info_calendar_event(self):
         return self.env['calendar.event'].\
             browse(self.reservation_event_id)
 
+    def do_clone_task_reservation(self):
+        if self.reservation_event_id:
+            self.get_calendar_event().write({'state': 'draft'})
+
+    def do_task_reservation(self):
+        self.draft_resources_reservation()
+        if self.task_state not in ['option', 'done']:
+            self.send_message('option')
+        self.write({'task_state': 'option'})
+        self.do_clone_task_reservation()
+
     @api.multi
     def do_reservation(self):
         self.ensure_one()
-        if self.activity_task_type == 'task':
-            self.draft_resources_reservation()
-            if self.task_state not in ['option', 'done']:
-                self.send_message('option')
         if self.is_activity():
             for child in self.child_ids:
-                child.draft_resources_reservation()
-                if child.task_state not in ['option', 'done']:
-                    child.send_message('option')
-                child.write({'task_state': 'option'})
+                child.do_task_reservation()
             self.send_message('option')
+            self.write({'task_state': 'option'})
+            return
```

Comment:
Since you already using an if statement and the function is not returning anything, an else seems better in this case. Don't you think?